File: Programmers/level2/cacheCities/cache_cities.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/17680
# 참고 블로그 : https://geonlee.tistory.com/102

def solution(cacheSize, cities):
    cache = []
    cnt = 0
    if cacheSize == 0:
        return len(cities) * 5
    while cities:
        city = cities.pop().lower()
        if city in cache:
            cache.remove(city)
            cache.append(city)
            cnt += 1
        else:
            if len(cache) < cacheSize:
                cache.append(city)
                cnt += 5
            else:
                del cache[0]
                cache.append(city)
                cnt += 5
    print(cache)
    print(cnt)
    return cnt

File: Programmers/level2/cacheCities/cacheCities.js
// https://programmers.co.kr/learn/courses/30/lessons/17680?language=javascript

export const solution = (cacheSize, cities) => {
    let answer = 0;
    let cache = [];
    if(cacheSize === 0){
        return cities.length * 5
    }
    cities.forEach(el => {
        el = el.toLowerCase();
        let idx = cache.indexOf(el);
        if(idx > -1){
            cache.splice(idx, 1);
            answer += 1;
        } else {
            if(cache.length >= cacheSize){
                cache.shift();
            }
            answer += 5;
        }
        cache.push(el);
    })
    return answer;
}